Change internal presense of the Grid, eliminating `_grid`

christianparpart opened the pull request:
This is the follow-up PR to PR #707.

shsms commented on src/frequenz/sdk/microgrid/grid.py:
Can't believe that this works, because now there's two `microgrid.grid` things - one a module, other a function.  If you remember, this was the reason we renamed the file to `_grid` originally.  Maybe you meant to move it to `timeseries.grid`?

christianparpart replied:
yeah, i'll adapt it, it is really interesting that the tests PASSED locally (!). I'm not done yet (apparently) :-)

christianparpart committed:
Change internal presense of the Grid, eliminating _grid

We are moving the grid to a new location to improve documentation
visibility of the type `Grid`.

This is not particularly easy, because we noticed that the
place where the grid module would fit the best (microgrid.grid),
would cause name conflicts. We agreed to move it to timeseries for now,
which is actually consistent to the current state, and move it further
later.

Signed-off-by: Christian Parpart <christian@parpart.family>

daniel-zullo-frequenz commented on src/frequenz/sdk/microgrid/_data_pipeline.py:
```diff
@@ -21,9 +21,9 @@
 from ..actor._actor import Actor
 from ..microgrid.component import Component
 from ..timeseries._grid_frequency import GridFrequency
+from ..timeseries.grid import Grid
+from ..timeseries.grid import get as get_grid
```
This can probably be done on the same line:
```suggestion
from ..timeseries.grid import get as get_grid, Grid
```

christianparpart replied:
`isort` will complain then. So I reverted that again.

daniel-zullo-frequenz reviewed (commented):
Only have an optional comment to check for and LGTM
